src/sas/qtgui/Plotting/Slicers/BoxSlicer.py

```diff
@@ -435,59 +610,296 @@ def clear(self):
+        """
+        self.save_x2 = self.x2
+        self.save_y2 = self.y2
+        self.save_x1 = self.x1
+        self.save_y1 = self.y1
+        self.save_half_height = self.half_height
+        self.save_half_width = self.half_width
+
+    def moveend(self, ev):
+        """
+        After a dragging motion update the 1D average plot and then reset the
+        flag self.has_move to False.
+        """
+        self.base.moveend(ev)
+        self.has_move = False
+
+    def restore(self, ev):
+        """
+        Restore the roughness for this layer.
+        """
+        self.y2 = self.save_y2
+        self.x2 = self.save_x2
+        self.y1 = self.save_y1
+        self.x1 = self.save_x1
+        self.half_height = self.save_half_height
+        self.half_width = self.save_half_width
+
+    def move(self, x, y, ev):
+        """
+        Process move to a new position, making sure that the move is allowed.
+        In principle, the move must not create a box without any data points
+        in it. For the dragging (continuous move), we make sure that the width
+        or height are not negative and that the entire ROI resides withing the
+        data. We leave the check of whether there are any data in that ROI to
+        the manipulations.py which is called from _post_data, itself being
+        called on moveend(ev).
+        """
+        if x - self.center_x > 0:
+            if self.center_x - (x - self.center_x) >= self.base.data.xmin:
+                self.x1 = x
+                self.half_width = self.x1 - self.center_x
+                self.x2 = self.center_x - self.half_width
+                self.has_move = True
+                self.base.update()
+                self.base.draw()
+            else:
+                logging.warning("The ROI must stay within the data please")
+        else:
+            logging.warning("you cannot go negative")
+
+    def setCursor(self, x, y):
+        """
+        Update the figure given x and y
+        """
+        self.move(x, y, None)
+        self.update()
+
+class HorizontalDoubleLine(BaseInteractor):
+    """
+    Draw 2 vertical lines that can move symmetrically in opposite directions in y and centered on
+    a point (PointInteractor). It also defines the left and right x positions of a box.
+    """
+    def __init__(self, base, axes, color='black', zorder=5, half_width=0.5, half_height=0.5,
+                 center_x=0.0, center_y=0.0):
+
+        BaseInteractor.__init__(self, base, axes, color=color)
+        # Initialization of the class
+        self.markers = []
+        self.axes = axes
+        # Center coordinates
+        self.center_x = center_x
+        self.center_y = center_y
+        # Box half width and height and  horizontal and vertical limits
+        self.half_height = half_height
+        self.save_half_height = self.half_height
+        self.half_width = half_width
+        self.save_half_width = self.half_width
+        self.y1 = self.center_y + half_height
```

Comment:
Humm... I think you mean that only the half_height is required? in principle, if moving horizontal lines neither the center nor the half_width should change. Regardless `restore` should restore everything that was saved in `save` which is what is done here.  That said I did look at this for this reason. It is however how these slicers were written back in the dawn of time and is exactly how `BoxSum` handles it as well (also since the dawn of time). Given all the surprising interactions encountered while working on this "small change" I opted not to go down that rabbit hole in this PR.  I think it would be worth placing a comment in the code for future work.

In terms of restoring, that is correct. The way this entire slicer module was designed (and I agree that it could definitely do with a refactor, but that is not what this PR is doing so we stay within the framework given) seems to be that restore function **only** changes the values back (restore) locally but that has to be followed with an update (which makes sure all the different actors are updated with those values, followed by a draw to actually render that on the screen for the user.  Remember that changing the horizontal line position (changing the height of the box) also mush change the length of the vertical lines.